src/System/src/FloatingBaseSystemDynamics.cpp

```diff
@@ -74,8 +74,25 @@ bool FloatingBaseDynamicalSystem::setMassMatrixRegularization(const iDynTree::Ma
     return true;
 }
 
-bool FloatingBaseDynamicalSystem::dynamics(const StateType& state,
-                                           const double& time,
+bool FloatingBaseDynamicalSystem::setState(const StateType& state)
+{
+
+    std::get<0>(m_state) = std::get<0>(state);
+    std::get<1>(m_state) = std::get<1>(state);
+    std::get<2>(m_state) = std::get<2>(state);
+    std::get<4>(m_state) = std::get<4>(state);
```

Comment:
I have to say that it is a little hard to remember what is what. I suspect there could be issues where we take one element for another. After a quick googling, there are some possible implementations of "tagged tuples", where the ``get`` uses a string instead of an index. Some references:
- https://stackoverflow.com/a/46862694
- https://gist.github.com/ilsken/dd91285d50197d6345f9
- https://github.com/duckie/named_types
- https://bitbucket.org/tlutz/named-variadic-template-proposal/src/master/
- https://stackoverflow.com/questions/13065166/c11-tagged-tuple

 Clearly, this is not related to this PR and probably there is no clean solution, and I don't know if the initializer lists would work anyway, but it may be worth discussing.